lib/modules/showImages.js

```diff
@@ -863,266 +856,186 @@ addModule('showImages', (module, moduleID) => {
 		updateParentHeight(expandoButton);
 	}
 
-	function generateImageExpando(expandoButton) {
-		const imageLink = expandoButton.imageLink;
-		const which = imageLink.galleryStart || 0;
-
-		const imgDiv = document.createElement('div');
-		imgDiv.classList.add('madeVisible');
-		imgDiv.currentImage = which;
-		imgDiv.sources = [];
-
-		// Test for a single image or an album/array of image
-		if (Array.isArray(imageLink.src)) {
-			imgDiv.sources = imageLink.src;
-
-			// Also preload images for an album if the option is on.
-			if (module.options.preloadImages.value) {
-				preloadImages(imageLink.src, 0);
-			}
-		} else {
-			// Only the image is left to display, pack it like a single-image album with no caption or title
-			const singleImage = {
-				src: imageLink.src,
-				href: imageLink.href
-			};
-			imgDiv.sources[0] = singleImage;
+	function generateMedia(options) {
+		switch (options.type) {
+			case 'GALLERY':
+				return generateGallery(options);
+				break;
+			case 'IMAGE':
+				return generateImage(options);
+				break;
+			//case 'TEXT':
+			//	generateTextExpando(expandoButton);
+			//	break;
+			//case 'IFRAME':
+			//	generateIframeExpando(expandoButton);
+			//	break;
+			case 'VIDEO':
+				return generateVideo(options);
+				break;
+			//case 'AUDIO':
+			//	generateAudioExpando(expandoButton);
+			//	break;
+			//case 'NOEMBED':
+			//	generateNoEmbedExpando(expandoButton);
+			//	break;
+			//case 'GENERIC_EXPANDO':
+			//	generateGenericExpando(expandoButton, imageLink.expandoOptions);
+			//	break;
+			default:
+				throw new Error(`Invalid expando type: ${options.type}`);
 		}
+	}
 
-		let header;
-		if ('imageTitle' in imageLink) {
-			header = document.createElement('h3');
-			header.classList.add('imgTitle');
-			$(header).safeHtml(imageLink.imageTitle);
-			imgDiv.appendChild(header);
-		}
+	async function generateGallery(options) {
+		const pieces = options.src.map(src => ({ src, container: null, media: null }));
 
-		if ('caption' in imageLink) {
-			const captions = document.createElement('div');
-			captions.className = 'imgCaptions';
-			$(captions).safeHtml(imageLink.caption);
-			imgDiv.appendChild(captions);
+		// No need for gallery functionality if length === 1
```

Comment:
We may still need to show title/description for gallery/album, but nav controls should be hidden.